tests/src/test/scala/cats/tests/ReducibleTests.scala

```diff
@@ -95,4 +95,37 @@ abstract class ReducibleCheck[F[_]: Reducible](name: String)(implicit ArbFInt: A
       fa.nonEmptyIntercalate(a) === (fa.toList.mkString(a))
     }
   }
+
+
+  test("Reducible#nonEmptyPartition retains size") {
+    forAll { (fi: F[Int], f: Int => Either[String, String]) =>
+      val folded = fi.nonEmptyPartition(f).fold(identity, identity, _ ++ _.toList)
+      folded.size.toLong should === (fi.size)
+    }
+  }
+
+  test("Reducible#nonEmptyPartition to one side is identity") {
+    forAll { (fi: F[Int], f: Int => String) =>
+      val g: Int => Either[Double, String] = f andThen Right.apply
+      val h: Int => Either[String, Double] = f andThen Left.apply
+
+      val withG = fi.nonEmptyPartition(g).right.getOrElse(NonEmptyList.one(""))
+      withG should === (Reducible[F].toNonEmptyList(fi).map(f))
+
+      val withH = fi.nonEmptyPartition(h).left.getOrElse(NonEmptyList.one(""))
+      withH should === (Reducible[F].toNonEmptyList(fi).map(f))
+    }
+  }
+
+  test("Reducible#nonEmptyPartition remains sorted") {
```

Comment:
this is a bit weird. The only `F[_]` part of it `toNonEmptyList`. After that, we are testing `nonEmptyPartition` on NonEmptyList, no?

Reply:
Yes that's true, I wrote this test when originally `nonEmptyPartition` was part of `NonEmptyList`. I could move the test there, if you'd like :) Otherwise, I'm not sure it's possible to sort an `F[_]` with just `Reducible`